apps/bfd-model/bfd-model-rif/src/main/resources/db/migration/V20241101111900986__create_samhsa_2_tables.sql

```diff
@@ -0,0 +1,119 @@
+DROP TYPE IF EXISTS tag_code CASCADE;
+CREATE TYPE tag_code AS ENUM('R', '42CFRPart2');
+
+CREATE TABLE IF NOT EXISTS ccw.hha_tags (
+    tag_id BIGINT,
+    clm_id BIGINT NOT NULL,
+    code TAG_CODE,
+    details JSONB,
+    PRIMARY KEY(tag_id),
+    CONSTRAINT fk_hha_tags_clm_id_to_hha_claims_clm_id
+        FOREIGN KEY (clm_id)
+            REFERENCES ccw.hha_claims(clm_id)
+    );
+CREATE INDEX IF NOT EXISTS hha_tags_clm_id_idx
+    ON ccw.hha_tags(clm_id);
+
+CREATE TABLE IF NOT EXISTS ccw.carrier_tags (
+    tag_id BIGINT,
+    clm_id BIGINT NOT NULL,
+    code TAG_CODE,
+    details JSONB,
+    PRIMARY KEY(tag_id),
+    CONSTRAINT fk_carrier_tags_clm_id_to_carrier_claims_clm_id
+        FOREIGN KEY (clm_id)
+            REFERENCES ccw.carrier_claims(clm_id)
+    );
+CREATE INDEX IF NOT EXISTS carrier_tags_clm_id_idx
+    ON ccw.carrier_tags(clm_id);
+
+CREATE TABLE IF NOT EXISTS ccw.dme_tags (
+    tag_id BIGINT,
+    clm_id BIGINT NOT NULL,
+    code TAG_CODE,
+    details JSONB,
+    PRIMARY KEY(tag_id),
+    CONSTRAINT fk_dme_tags_clm_id_to_dme_claims_clm_id
+        FOREIGN KEY (clm_id)
+            REFERENCES ccw.dme_claims(clm_id)
+    );
+CREATE INDEX IF NOT EXISTS dme_tags_clm_id_idx
+    ON ccw.dme_tags(clm_id);
+
+CREATE TABLE IF NOT EXISTS ccw.hospice_tags (
+    tag_id BIGINT,
+    clm_id BIGINT NOT NULL,
+    code TAG_CODE,
+    details JSONB,
+    PRIMARY KEY(tag_id),
+    CONSTRAINT fk_hospice_tags_clm_id_to_hospice_claims_clm_id
+        FOREIGN KEY (clm_id)
+        REFERENCES ccw.hospice_claims(clm_id)
+    );
+CREATE INDEX IF NOT EXISTS hospice_tags_clm_id_idx
+    ON ccw.hospice_tags(clm_id);
+
+CREATE TABLE IF NOT EXISTS ccw.outpatient_tags (
+    tag_id BIGINT,
+    clm_id BIGINT NOT NULL,
+    code TAG_CODE,
+    details JSONB,
+    PRIMARY KEY(tag_id),
+    CONSTRAINT fk_outpatient_tags_clm_id_to_outpatient_claims_clm_id
+        FOREIGN KEY (clm_id)
+            REFERENCES ccw.outpatient_claims(clm_id)
+    );
+CREATE INDEX IF NOT EXISTS outpatient_tags_clm_id_idx
+    ON ccw.outpatient_tags(clm_id);
+
+CREATE TABLE IF NOT EXISTS ccw.snf_tags (
+    tag_id BIGINT,
+    clm_id BIGINT NOT NULL,
+    code TAG_CODE,
+    details JSONB,
+    PRIMARY KEY(tag_id),
+    CONSTRAINT fk_snf_tags_clm_id_to_snf_claims_clm_id
+        FOREIGN KEY (clm_id)
+            REFERENCES ccw.snf_claims(clm_id)
+    );
+CREATE INDEX IF NOT EXISTS snf_tags_clm_id_idx
+    ON ccw.snf_tags(clm_id);
+
+CREATE TABLE IF NOT EXISTS ccw.inpatient_tags (
+    tag_id BIGINT,
+    clm_id BIGINT NOT NULL,
+    code TAG_CODE,
+    details JSONB,
+    PRIMARY KEY(tag_id),
+    CONSTRAINT fk_inpatient_tags_clm_id_to_inpatient_claims_clm_id
+        FOREIGN KEY (clm_id)
+            REFERENCES ccw.inpatient_claims(clm_id)
+    );
+CREATE INDEX IF NOT EXISTS inpatient_tags_clm_id_idx
+    ON ccw.inpatient_tags(clm_id);
+
+CREATE TABLE IF NOT EXISTS rda.mcs_tags (
+    tag_id BIGINT,
+    clm_id CHARACTER VARYING(15) NOT NULL,
```

Comment:
any reason `clm_id` data type is `BIGINT` for ccw schema and CHARACTER VARYING(15) for rda?

Reply:
Yes, it has to do with the foreign keys. mcs_claims and fiss_claims use `CHARACTER VARYING` for the claim id type, which necessitates the corresponding key in the tags tables being the same.